libyul/backends/evm/ControlFlowGraph.h

```diff
@@ -0,0 +1,199 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+/**
+ * Control flow graph and stack layout structures used during code generation.
+ */
+
+#pragma once
+
+#include <libyul/AST.h>
+#include <libyul/AsmAnalysisInfo.h>
+#include <libyul/Dialect.h>
+#include <libyul/Scope.h>
+
+#include <functional>
+#include <list>
+#include <vector>
+
+namespace solidity::yul
+{
+
+/// The label pushed as return label before a function call, i.e. the label the call is supposed to return to.
+struct FunctionCallReturnLabelSlot
+{
+	std::reference_wrapper<yul::FunctionCall const> call;
+	bool operator==(FunctionCallReturnLabelSlot const& _rhs) const { return &call.get() == &_rhs.call.get(); }
+	bool operator<(FunctionCallReturnLabelSlot const& _rhs) const { return &call.get() < &_rhs.call.get(); }
```

Comment:
The order is indeed nondeterministic, but it's only needed them to construct ``set``s and ``map``s of stack slots, none of which are traversed, so the order won't affect the result. It would of course be nicer to have a deterministic order, but it's nontrivial to construct one.

Reply:
ok. Seems fine then.